include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -216,7 +216,8 @@ struct transform_reduce_known
                const ::std::size_t __global_offset, _AccLocal& __local_mem, const _Acc&... __acc) const
     {
         const ::std::size_t __adjusted_global_id = __global_offset + __iters_per_work_item * __global_id;
-        const ::std::int64_t __items_to_process = __n - (__iters_per_work_item * __global_id);
+        const ::std::int64_t __items_to_process =
+            static_cast<std::int64_t>(__n) - static_cast<std::int64_t>(__iters_per_work_item * __global_id);
```

Comment:
Technically, casting `size_t` to `int64_t` on 64-bit paltforms is a downcast and may lose significant digits :)

I suggest to take a slightly different approach: compute the number of work items that will have work  (`_n / __iters_per_work_item`) , and then compare it to the ID of the item.

Reply:
Good point. I think we can work around the negative numbers. I will send out a patch shortly.

Reply:
While this is technically true, if it is any performance cost to support sizes of n greater than 9,223,372,036,854,775,807 (maximum signed 64 bit integer), I don't think it is worth it.

Reply:
I have just pushed a change to first check if we can do the full `__iters_per_work_item` loop and if not if the work item has work at all. If so, we calculate `__items_to_process`. Please have another look.

Reply:
Yes, this is an improvement and should be at least as good performance wise.

Reply:
This change looks good to me as well.

Reply:
What is under review right now looks much better comparing with the previous version. Need to go though it logically one more time (for me)